kong/api/routes/health.lua

```diff
@@ -14,6 +15,20 @@ local kong = kong
 
 
 return {
+  ["/config/ready"] = {
+    GET = function(self, dao, helpers)
+      if kong.db.strategy ~= "off" then
+        return kong.response.exit(200)
+      end
+    -- unintuitively, "true" is unitialized. we do always initialize the shdict key
+    -- after a config loads, this returns the hash string
+    if declarative.get_current_hash() == true then
```

Comment:
This bothers me a bit. Especially as I saw that in `LMDB` branch it becomes `"true"`. @dndx did you change the `true` there to something zeroes hash `0000000` or something? To make this detail leak less, what do you think @rainest, should we have this detail at least closer to that `declarative` file? I mean something like:

```
```suggestion
    if declarative.has_config() then
```

Reply:
7054e4e for moving the logic. What's the other branch in question? I didn't think we'd have any reason to change that default, and even though `true` looked like an odd choice, I left it as-is for lack of a compelling reason to change it. Since it is now used in multiple sections of the codebase, it seemed prudent to move it to a constant: 
0d947c5

Reply:
@rainest here it is changed: https://github.com/Kong/kong/pull/8224/files#diff-aac904bbcdee399964e000cfef300d7f3417d18b23dea86d192b4a42e2fcbfeeR819-R820

That is still not merged though.

Reply:
Ah, hmm, this will interact with that then, we probably should hold this until after that PR is complete.